### DISCUSSION TITLE
Deploying a pre-built Next.js app

### DISCUSSION THREAD

**Discussion by timswalling** [vercel/vercel#4404] [Help] [answered]
We're using a Nx monorepo to develop several Next.js apps with a large amount of shared code.

We've managed to configure multiple deployments (see zeit/now#4384), but in our workflow the build step feels redundant as we're already passing environment variables and installing dependencies during CI.

Is there a way to deploy a pre-built Next.js app? I can see that the `now-next` builder takes care of route mapping for the various static files and lambda functions, but it appears to need to be run on the original (uncompiled) project.

**Comment by paulogdm** [comment, marked as answer]
At the moment, we don't plan to support "pre-built" Next.js apps since we do need to control the generated output and underlying routing. So no, it is not possible, and we don't envision a future where this would be implemented.

On the other hand, Monorepo support is close, and it will bring more clarity to complex setups. Also, do remember that you can run any build command that you want in your [project's settings](https://vercel.com/blog/advanced-project-settings)

**Reply by timswalling** [reply]
Thanks for the reply.

We've managed to get deployments working fine by running the build using the appropriate `nx build` command. I guess it's just a bit annoying that we install all the dependencies in our CI to run tests prior to deploying, then have to wait again while they are installed during deployment.
